programmers/su-park-su.js

```diff
@@ -0,0 +1,19 @@
+function solution(n){
+    let A = "수박";
+    let B = "수";
+    let C = Math.floor(n/2);
+    let D ="";
+    if (n == 1){
+        return console.log("수");
+    }
+    for (let i = 0 ;  i<C ; i++){
+        D+=A;
+    }
+    if ( n % 2 !=0){
+        return (D+B);
+    }
+    else {
+        return (D);
+    }
+}
+console.log(solution(3));
```

Comment:
위의 코드를 보면 고칠 점이 있습니다.
1. 변수의 선언을 어떻게 해야 되는지 잘 인지하시지 못하고 있습니다.
2. 변수이름을 너무 대충 지으시고 계십니다.

let 선언은 변수의 값이 코드의 흐름내에서 변하는 값이라면 let으로 선언을 해줍니다.
변하지 않는 값이라면 const로 선언을 해줍니다.
변수 이름같은 경우도 const로 선언해주었을때 ``` 상수 ```라면 대문자로 변수명을 지어줍니다. 띄어쓰기는 _ 로 구분해줍니다.

if (n % 2 !== 0) 같은 경우에도 n % 2가 짝수라면 0  홀수라면 1이 나올 것입니다.
자바스크립트에서는 숫자 0은 false 1은 true이기때문에 if (n % 2)로 쓰셔도 됩니다.

```suggestion
function solution(n) {
    const SU = "수";
    const SUBAK = "수박";
    const CONTENT_LENGTH = Math.floor(n / 2);
    let result = "";
    
    if (n == 1){
        return SU;
    }
    
    for (let i = 0; i < CONTENT_LENGTH ; i++){
        result += SUBAK;
    }
    
    if (n % 2) {
        return result + SU;
    } 
    return result;
}
```